.github/workflows/calc-cov-summaries.yml

```diff
@@ -0,0 +1,59 @@
+# Reusable workflow to calculate coverage; don't push to code cov.
+# instead, use octocov, which allows all data to remain on github.
+
+name: calc-cov-summaries
+
+on:
+  workflow_call:
+    inputs:
+      use-public-rspm:
+        required: false
+        type: boolean
+        default: true
+
+# limits permisions to write, as the common use case is for pull requests;
+# this means certain options may not work, but makes the workflow more secure
+permissions:
+  pull-requests: write
+
+jobs:
+  calc-cov-summaries:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          use-public-rspm: ${{ inputs.use-public-rspm }}
+
+
+      # need: coverage allows users to put additional pkgs for coverage calculations
+      # into the description file under a field called Config/Needs/coverage
+      # For more info, see https://r-pkgs.org/dependencies-in-practice.html#sec-dependencies-nonstandard-config-needs
+      - uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          extra-packages: any::covr, any::xml2
+          needs: coverage
+
+      - name: Calc. Test coverage
+        run: |
+          cov <- covr::package_coverage(
+            quiet = FALSE,
+            clean = FALSE,
+            install_path = file.path(normalizePath(Sys.getenv("RUNNER_TEMP"), winslash = "/"), "package")
+          )
+          print(cov)

```

Comment:
[Non-blocking suggestion]: Maybe generate a standalone HTML report with `report <- covr::report(browse = FALSE, file = "covr_report.html")` and uploading it to the GHA run. This would provide detailed line-by-line coverage.

Reply:
Great idea! I will add this.